### DISCUSSION TITLE
Using authentication method not supported by ApiDefinition CRD

### DISCUSSION THREAD

**Discussion by omrilama** [TykTechnologies/tyk-operator#396] [Q&A] [answered]
Hi,

I installed the tyk gateway OSS (using helm) and tyk-operator on my k8s cluster and just noticed that HMAC is yet to be supported by the ApiDefinition CRD.

I was wondering if, when using the tyk-operator, I must define all definitions using the ApiDefinition or can I also use the dashboard or simply the gateway api (https://tyk.io/docs/tyk-gateway-api/api-definition-objects/) ? 

Assuming I can use either (the dashboard or the api-gateway api) I guess I can support HMAC but it just won't be represented in the ApiDefintion CRD?

Is there any other/better option for me to support HMAC and still use oss tyk?

Thanks!

**Comment by asoorm** [comment, marked as answer]
Hi, and thanks for your enquiry. We didn't implement HMAC as an authentication method yet for Tyk Operator - because we did not see demand for it till now and we had to prioritise more popular features.

![Screenshot 2022-02-15 at 13 17 25](https://user-images.githubusercontent.com/1465130/154069745-9a330cda-e3ea-4e9b-aa3b-872d73f672b5.png)

[API Definition CRD Documentation](https://github.com/TykTechnologies/tyk-operator/blob/master/docs/api_definitions.md)

If this is something critical for you - please open a feature request (GH issue) and we can bump the priority for you.

[Open a feature request](https://github.com/TykTechnologies/tyk-operator/issues/new?assignees=&labels=enhancement&template=FEATURE.md)

**Reply by omrilama** [reply]
Thank you for the reply.

Is it possible though to use the gateway api (or some other way) to create this authentication (without a CRD) assuming I'm running on the open source version?
